ui/app/AppLayouts/Profile/views/PasswordView.qml

```diff
@@ -0,0 +1,356 @@
+            // * Password is in pwnd passwords database
+            if(isInPwndDatabase())
+                errorTxt.text = qsTr("This password has been pwned and shouldn't be used")
+
+            // * Common password
+            else if(isCommonPassword())
+                errorTxt.text = qsTr("This password is a common word and shouldn't be used")
+
+            // * Password too short
+            else if(isTooShort())
+                errorTxt.text = qsTr("Password must be at least 6 characters long")
+        }
+
+        function isInPwndDatabase() {
+            console.warn("TODO: Pwnd password validation NOT implemented yet! ")
+            return false
+        }
+
+        function isCommonPassword() {
+            console.warn("TODO: Common password validation NOT implemented yet! ")
+            return false
+        }
+
+        function isTooShort() { return newPswInput.text.length < root.minPswLen }        
+    }
+
+    spacing: 3 * Style.current.padding / 2
+    z: root.zFront
+    width: 416
+
+    // View visual content:
+    StatusBaseText {
+        id: title
+        anchors.horizontalCenter: parent.horizontalCenter
+        visible: root.titleVisible
+        text: root.title
+        font.pixelSize: 22
+        font.bold: true
+        color: Theme.palette.directColor1
+    }
+
+    Column {
+        StatusBaseText {
+            id: introTxt
+            anchors.horizontalCenter: parent.horizontalCenter
+            text: root.introText
+            font.pixelSize: 12
+            color: Theme.palette.baseColor1
+        }
+
+        StatusBaseText {
+            id: recoverTxt
+            anchors.horizontalCenter: parent.horizontalCenter
+            text: root.recoverText
+            font.pixelSize: 12
+            color: Theme.palette.dangerColor1
+        }
+    }
+
+    // TODO replace with StatusInput as soon as it supports password
+    Input {
+        id: currentPswInput
+        z: root.zFront
+        visible: !root.createNewPsw
+        width: parent.width
+        placeholderText: qsTr("Current password")
+        textField.echoMode: TextInput.Password
+        textField.validator: d.validator
+        keepHeight: true
+        textField.rightPadding: showHideCurrentIcon.width + showHideCurrentIcon.anchors.rightMargin + Style.current.padding / 2
+
+        StatusFlatRoundButton {
+            id: showHideCurrentIcon
+            visible: currentPswInput.text !== ""
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.right: parent.right
+            anchors.rightMargin: 16
+            width: 24
+            height: 24
+            icon.name: currentPswInput.textField.echoMode == TextInput.Password ? "show" : "hide"
+            icon.color: Theme.palette.baseColor1
+
+            onClicked: currentPswInput.textField.echoMode == TextInput.Password ? currentPswInput.textField.echoMode = TextInput.Normal : currentPswInput.textField.echoMode = TextInput.Password
```

Comment:
Same thing as previously stated for the `echoMode`. Maybe even we want to put that input in a separate component since it's used in both places?
or it's what you meant with the comment above that this will all be handled by the StatusQ component?

Reply:
Yes! There is a [task](https://github.com/status-im/StatusQ/issues/393) for including the `echoMode` into `StatusInput` and I think it also makes sense to integrate the hide/show icon button there which could be shown or hidden depending on the user need.